drivers/overlay/joinleave.go

```diff
@@ -72,7 +87,7 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		}
 	}
 
-	d.peerDbAdd(nid, eid, ep.addr.IP, ep.mac,
+	d.peerDbAdd(nid, eid, ep.addr.IP, ep.addr.Mask, ep.mac,
```

Comment:
@mrjana peerKey structure has only the peerIP. I added the mask to peerEntry. That is because on a neighbor miss notification we only get the IP. So in the peerDB we can't change it to IPNet. It can be changed in the serf exchange. But it doesn't buy us much. WDYT ?

I pushed the changes with other comments addressed. PTAL